coreax/util.py

```diff
@@ -244,6 +245,41 @@ def solve_qp(kernel_mm: ArrayLike, gramian_row_mean: ArrayLike, **osqp_kwargs) -
     return sol.primal
 
 
+def sample_batch_indices(
+    random_key: KeyArrayLike,
+    data_size: int,
+    batch_size: int,
+    num_batches: int,
+) -> ArrayLike:
+    """
+    Sample an array of indices of size batch_size x num_batches.
+
+    Each column (batch) of the sampled array will contain unique elements. The largest
+    possible index sampled is dictated by data_size.
+
+    :param random_key: Key for random number generation
+    :param data_size: Size of the data we wish to sample from
+    :param batch_size: Size of the batch we wish to sample
+    :param num_batches: Number of batches to sample
+    :return: Array of batch indices of size `batch_size` x `num_batches`
+    """
+    if data_size < batch_size:
```

Comment:
I believe the approach we have been taking so far is to only raise custom messages when the underlying code doesn't raise anything helpful/obvious to the user, and to be as permissive as possible.

By this I mean that, for example: if `data_size=1.0`, it will not pass the isinstance check, but it would pass conversion to an int, `int(1.0)`. While the type hint restricts to integers, we can reasonably accept any value that can be cast to an integer. The caveat here is that an 'invalid' value of say '1.5' will be silently cast to the integer `1.0`, which may be unexpected. However, the type hint is already indicating that behaviour is undefined for anything that isn't an `int`, we're just trying to be as lenient as possible. This is the same reason why we use try-except in many places, over explicit type checks.

Reply:
Makes sense, removed the unnecessary checks. Left the ones in that don't raise errors but behave in a weird way. Removed the corresponding tests